src/Microsoft.ML.Data/DataView/InternalSchemaDefinition.cs

```diff
@@ -192,7 +192,12 @@ public static void GetVectorAndItemType(string name, Type rawType, IEnumerable<A
             // Check if the itemType extracted from rawType is supported by ML.NET's type system.
             // It must be one of either ML.NET's pre-defined types or custom types registered by the user.
             else if (!itemType.TryGetDataKind(out _) && !DataViewTypeManager.Knows(itemType, attributes))
-                throw Contracts.ExceptParam(nameof(rawType), "Could not determine an IDataView type for member {0}", name);
+            {
+                //This is a big hack. It gives helpful information when customer accidently define OnnxSequenceType the same as member type
+                if (DataViewTypeManager.Knows(typeof(IEnumerable<>).MakeGenericType(itemType), attributes))
```

Comment:
Can you please explain why this is a big hack?

Reply:
Because I use DataViewTypeManager to check if there's "make-up" types in it(e.g. customer define the type "float", and this will check if "iEnumerable<float>" in the DateTypeManager)

Reply:
Can you give some examples why the suggestion may not be correct? Is it possible to capture in what cases we can give accurate suggestions?

---
In reply to: [423232559](https://github.com/dotnet/machinelearning/pull/5114#discussion_r423232559) [](ancestors = 423232559)